Doc/library/typing.rst

```diff
@@ -1177,38 +1191,65 @@ These can be used as types in annotations using ``[]``, each having a unique syn
    * Multiple type annotations are supported (``Annotated`` supports variadic
      arguments)::
 
-       Annotated[int, ValueRange(3, 10), ctype("char")]
+        @dataclass
+        class ctype:
+            kind: str
+
+        Annotated[int, ValueRange(3, 10), ctype("char")]
 
    * ``Annotated`` must be called with at least two arguments (
      ``Annotated[int]`` is not valid)
 
    * The order of the annotations is preserved and matters for equality
      checks::
 
-       Annotated[int, ValueRange(3, 10), ctype("char")] != Annotated[
-           int, ctype("char"), ValueRange(3, 10)
-       ]
+        assert Annotated[int, ValueRange(3, 10), ctype("char")] != Annotated[
+            int, ctype("char"), ValueRange(3, 10)
+        ]
 
    * Nested ``Annotated`` types are flattened, with metadata ordered
      starting with the innermost annotation::
 
-       Annotated[Annotated[int, ValueRange(3, 10)], ctype("char")] == Annotated[
-           int, ValueRange(3, 10), ctype("char")
-       ]
+        assert Annotated[Annotated[int, ValueRange(3, 10)], ctype("char")] == Annotated[
+            int, ValueRange(3, 10), ctype("char")
+        ]
 
    * Duplicated annotations are not removed::
 
-       Annotated[int, ValueRange(3, 10)] != Annotated[
-           int, ValueRange(3, 10), ValueRange(3, 10)
-       ]
+        assert Annotated[int, ValueRange(3, 10)] != Annotated[
+            int, ValueRange(3, 10), ValueRange(3, 10)
+        ]
+
+   * ``Annotated`` can be used with nested and generic aliases:
+
+     .. testcode::
 
-   * ``Annotated`` can be used with nested and generic aliases::
+        T = TypeVar('T')
 
-       T = TypeVar('T')
-       Vec = Annotated[list[tuple[T, T]], MaxLen(10)]
-       V = Vec[int]
+        @dataclass
+        class MaxLen:
+            value: int
+
+        Vec = Annotated[list[tuple[T, T]], MaxLen(10)]
+        V = Vec[int]
+
+        assert V == Annotated[list[tuple[int, int]], MaxLen(10)]
```

Comment:
I briefly tried to convert this example to use a PEP-695 type statement, but it's nontrivial to do so, since PEP-695 type statements don't seem to do TypeVar substitution in the same way. (I might open an issue about this; it seems like a flaw in the implementation.)

```pycon
>>> from typing import Annotated
>>> type Vec[T] = Annotated[list[tuple[T, T]], "metadata"]
>>> type V = Vec[int]
>>> V.__value__
Vec[int]
>>> type(V.__value__)
<class 'types.GenericAlias'>
>>> V.__value__.__value__
typing.Annotated[list[tuple[T, T]], 'metadata']
>>> # How do we get to `typing.Annotated[list[tuple[int, int]], 'metadata']`?
```

Reply:
That's right, `TypeAliasType.__getitem__` doesn't attempt to do parameter substitution. That was an intentional choice. I'd rather leave resolving typevars to type checkers.

Reply:
I would suggest using PEP 695 syntax and instead of an explicit assert, say that type checkers should treat it equivalently to the expanded version.

Reply:
Done